sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosPatch.java

```diff
@@ -0,0 +1,198 @@
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Grammar based on RFC: https://tools.ietf.org/html/rfc6902#section-4.1
+ *
+ * Contains a list of Patch operations to be applied on an item. It is applied in an atomic manner and the operations
+ * grammar follows above RFC.
+ *
+ * This can be executed in 3 ways:
+ *  1. Passing this to container in container.patchItem() which requires the id of the item to be patched, partition
+ *      key, the CosmosPatch instance, any CosmosItemRequestOptions and the class type for which response will be parsed.
+ *  2. Add CosmosPatch instance in TransactionalBatch using batch.patchItemOperation() which requires the id of the item
+ *      to be patched, cosmos patch instance and TransactionalBatchItemRequestOptions(if-any) and follow remaining
+ *      steps for batch for it's execution.
+ *  3. Create a bulk item using BulkOperations.getPatchItemOperation which requires the id of the item to be patched,
+ *      cosmos patch instance, partition key and BulkItemRequestOptions(if-any) and follow remaining steps to
+ *      execute bulk operations.
+ */
+@Beta(Beta.SinceVersion.V4_9_0)
+public final class CosmosPatch {
+
+    private final List<PatchOperation> patchOperations;
+
+    private CosmosPatch() {
+        this.patchOperations = new ArrayList<>();
+    }
+
+    /**
+     * Initializes a new instance of {@link CosmosPatch} that will contain operations to be performed on a item atomically.
+     *
+     * @return A new instance of {@link CosmosPatch}.
+     */
+    @Beta(Beta.SinceVersion.V4_9_0)
+    public static CosmosPatch create() {
+        return new CosmosPatch();
+    }
+
+    /**
+     * This performs one of the following functions, depending upon what the target location references:
+     *  1. Target location specifies an array index, a new value is inserted into the array at the specified index.
+     *  2. Target location specifies an object member that does not already exist, a new member is added to the object.
+     *  3. Target location specifies an object member that does exist, that member's value is replaced.
+     *
+     * @param <T> The type of item to be added.
+     *
+     * @param path the operation path.
+     * @param value the value which will be added.
+     *
+     * @return same instance of {@link CosmosPatch}
+     */
+    @Beta(Beta.SinceVersion.V4_9_0)
+    public <T> CosmosPatch add(String path, T value) {
+
+        checkNotNull(value, "expected non-null value");
+        checkArgument(StringUtils.isNotEmpty(path), "path empty %s", path);
+
+        this.patchOperations.add(
+            new PatchOperationCore<>(
+                PatchOperationType.ADD,
+                path,
+                value));
+
+        return this;
+    }
+
+    /**
+     * This removes the value at the target location.
+     *
+     * @param path the operation path.
```

Comment:
It makes sense to reference the RFC for the details - but I would expect as a developer to see a brief sample for the common options on how a correct "path" value looks like - like how to reference a json property - a nested property - and array value etc. I think a couple of samples in the javadoc would be really helpful.

Reply:
Have added patch examples based on a json object mentioned at the class level. This should help. Will take a confirmation from store team before merging for those docs.